File: BackEnd/Application/UseCases/Projects/UpdateProject/UpdateProjectCommandHandler.cs
using Domain.Aggregates;
using Infrastructure.Repository;
using Infrastructure.UnitOfWork;
using MediatR;

namespace Application.UseCases.Projects.UpdateProject;

public class UpdateProjectCommandHandler(IRepository<Project, Guid> projectRepository, IUnitOfWork unitOfWork) : IRequestHandler<UpdateProjectCommand, bool>
{
    public async Task<bool> Handle(UpdateProjectCommand request, CancellationToken cancellationToken)
    {
        var project = await projectRepository.GetByIdAsync(request.Id, cancellationToken);
        if (project == null) return false;
        project.Update(request.Name, request.Description, request.Code, request.StartDate, request.EndDate);
        await projectRepository.UpdateAsync(project);
        await unitOfWork.SaveChangesAsync(cancellationToken);
        return true;
    }
}

File: flutter/lib/application/use_cases/projects/update_project/update_project_command_handler.dart
import 'package:enterprise_management/application/use_cases/projects/update_project/update_project_command.dart';
import 'package:enterprise_management/infrastructure/repositories/project_repository.dart';
import 'package:enterprise_management/shared_kernel/cqrs/command_handler.dart';
import 'package:enterprise_management/shared_kernel/result/result.dart';

class UpdateProjectCommandHandler extends ICommandHandler<UpdateProjectCommand, bool> {
  const UpdateProjectCommandHandler({required this._projectRepository});
  final IProjectRepository _projectRepository;
  @override
  Future<Result<bool>> handle(UpdateProjectCommand command) async {
    final res = await _projectRepository.updateProject(id: command.id, name: command.name, description: command.description ,code: command.code);
    return Result.success(res);
  }

}